org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/jvmmodel/DomainmodelJvmModelInferrer.xtend

```diff
@@ -87,8 +89,14 @@ class DomainmodelJvmModelInferrer extends AbstractModelInferrer {
 		inferredType.handleDuplicateJvmOperations[jvmOperations|
 			// we only remove getters/setters we created automatically
 			val getterOrSetter = jvmOperations.filter[primarySourceElement instanceof Property].head
-			if (getterOrSetter !== null)
+			if (getterOrSetter !== null) {
+				val property = getterOrSetter.primarySourceElement as Property
+				removeAssociation(property, getterOrSetter)
+				removeAssociation(property.type, getterOrSetter.returnType)
```

Comment:
is there a change to avoid two calls?

Reply:
Is there a chance to avoid adding the bad setters / getters in the first place?

Reply:
For setters, it would be the parameter type, wouldn't it?

Reply:
> Is there a chance to avoid adding the bad setters / getters in the first place?

I was wondering that myself!
I think I worked on that part (though I don't remember I committed those removal operations).

Reply:
will check for setters

Reply:
[Foo.getB() in context of Foo, Foo.setB(java.lang.String) in context of Foo, Foo.setB(void) in context of Foo]
looks like the setters are not really found as duplicate

Reply:
fixed